### ISSUE TITLE
Early return from inside `for` loop fails

### ISSUE THREAD

**Issue by sysread** [abs-lang/abs#303]
```
test = f() {
    for i in 1..10 {
        if i == 5 {
            return false
        }
    }

    return true
}

echo(test())
```

Expected result: `false`
Actual result: `true`

**Comment by odino** [comment]
This is a nasty regression that should be fixed soon -- early `return` is a very convenient feature.

**Comment by odino** [comment]
@sysread this is going to get fixed today / tomorrow

**Comment by odino** [comment]
This is fixed in `1.10.1`
